### ISSUE TITLE
how to extract entire text for given image using Donut pretrained model?

### ISSUE THREAD

**Issue by alenma05** [clovaai/donut#62]

**Comment by satheeshkatipomu** [comment]
I think it is straightforward to get text output using donut-base model. Load  `naver-clova-ix/donut-base`  from huggingface and use `<s_synthdog>` as prompt.
```
from donut import DonutModel
import torch
from PIL import Image


pretrained_model = DonutModel.from_pretrained("naver-clova-ix/donut-base")
if torch.cuda.is_available():
    pretrained_model.half()
    device = torch.device("cuda")
    pretrained_model.to(device)
else:
    pretrained_model.encoder.to(torch.bfloat16)
pretrained_model.eval()

task_name = "synthdog"
task_prompt = f"<s_{task_name}>"

input_img = Image.open("./test_images/example1.png")
output = pretrained_model.inference(image=input_img, prompt=task_prompt)["predictions"][0]
print(output)
```

**Comment by alenma05** [comment]
Thank you for your help.

**Comment by yagcaglar** [comment]
Hi, thanks for the repo and the code.
When I use this code in my dataset, I got dummy text from some of the documents such as
RT @ RT @ RT @ RT @ RT @ RT @ RT @ RT @ RT @ or
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 or
(  ) (  )  (  )  (  )  (  )  (  )  (  )  (  )  or
| | | | | | | | | | | | | | | | | | | | | 
where each output is from different images. Might it be related to resolution of the input? Can I change something to prevent this?

**Comment by gamingflexer** [comment]
<img width="1024" alt="Screenshot 2023-01-07 at 12 54 26 PM" src="https://user-images.githubusercontent.com/50608241/211139227-1376396d-8e7b-463a-8b98-f1e79374dc9b.png">
 Getting this issue

**Comment by llStringll** [comment]
Your model is running in fp16 half precision floating points, but the image you passed is a full precision float. Either change the model to run on full precision or change the image to half precision floating point numpy array

**Comment by swarnalatapatra** [comment]
I am getting attribute error . can anyone please suggest ?
<img width="679" alt="image" src="https://github.com/clovaai/donut/assets/45567398/60e2c3f5-74e9-454d-b0c1-814cbcc9378f">

**Comment by llStringll** [comment]
Can you share initialisation of pretrained_model variable please, also if possible, are you using this repo's implementation over SwinTransformer or Huggingface's?

**Comment by swarnalatapatra** [comment]
I am using the below for pretrained model where task_name is docvqa

parser = argparse.ArgumentParser()
parser.add_argument("--task", type=str, default="docvqa")
parser.add_argument("--pretrained_path", type=str, default="naver-clova-ix/donut-base")
args, left_argv = parser.parse_known_args()

task_name = args.task
if "docvqa" == task_name:
    task_prompt = "<s_docvqa><s_question>{user_input}</s_question></s_docvqa>"
else:  # rvlcdip, cord, ...
    task_prompt = f"<s_{task_name}>"

pretrained_model = DonutModel.from_pretrained(args.pretrained_path, ignore_mismatched_sizes=True)

if torch.cuda.is_available():
    pretrained_model.half()
    device = torch.device("cuda")
    pretrained_model.to(device)
else:
    pretrained_model.encoder.to(torch.float32)

pretrained_model.eval()

**Comment by llStringll** [comment]
What is the version of timm that you are using? It is inconsistent with this repo, specifically SwinTransformer class from timm module is no longer having an attribute pos_drop, a dropout for position embeddedings...Please try it with a consistent version of timm, as per this repo's readme.

**Comment by llStringll** [comment]
If you refer [here](https://github.com/huggingface/pytorch-image-models/tree/main#april-12-2023)
see, just last month they refactored these attributes, definitely including pos_drop argument....and your timm version is definitely very recent....downgrade it to something like 0.5.4, and it should work pretty fine.

**Comment by aindilis** [comment]
> Your model is running in fp16 half precision floating points, but the image you passed is a full precision float. Either change the model to run on full precision or change the image to half precision floating point numpy array

I am new to Python, but after much tinkering was unable to figure out how to change the image to half precision floating point numpy array. I would greatly appreciate if someone could please post the code to make the conversion here.
